app/src/main/java/com/compscieddy/meetinthemiddle/MapsActivity.java

```diff
@@ -153,95 +161,120 @@ public void onLocationChanged(Location location) {
+    public void onMapReady(GoogleMap googleMap) {
+        mMap = googleMap;
+        try {
+            // "MyLocation" is the "blue dot" feature for showing the current location and jumping to the location
 //      mMap.setMyLocationEnabled(true);
-    } catch (SecurityException se) {
-      lawg.e("se: " + se);
+
+        } catch (SecurityException se) {
+            lawg.e("se: " + se);
+        }
+
+        mMap.setOnMapClickListener(this);
+        mMap.setOnMapLongClickListener(this);
     }
-  }
 
-  @Override
-  public void onConnected(@Nullable Bundle bundle) {
-    try {
-      mLastLocation = LocationServices.FusedLocationApi.getLastLocation(mGoogleApiClient);
-      if (mLastLocation != null) {
-        double latitude = mLastLocation.getLatitude();
-        double longitude = mLastLocation.getLongitude();
-        LatLng latLng = new LatLng(latitude, longitude);
-        mLastKnownCoord.set(latitude, longitude);
-        if (mCurrentMarker != null) mCurrentMarker.remove();
+    @Override
+    public void onConnected(@Nullable Bundle bundle) {
+        try {
+            mLastLocation = LocationServices.FusedLocationApi.getLastLocation(mGoogleApiClient);
+            if (mLastLocation != null) {
+                double latitude = mLastLocation.getLatitude();
+                double longitude = mLastLocation.getLongitude();
+                LatLng latLng = new LatLng(latitude, longitude);
+                mLastKnownCoord.set(latitude, longitude);
+                if (mCurrentMarker != null) mCurrentMarker.remove();
+
+                Bitmap icon = BitmapFactory.decodeResource(getResources(),
+                        R.drawable.ic_darren);
+
+                Bitmap croppedIcon = getCroppedBitmap(icon);
+
+                mCurrentMarker = mMap.addMarker(new MarkerOptions().position(latLng).title("Current Location").icon(BitmapDescriptorFactory.fromBitmap(croppedIcon)));
+
+            }
+        } catch (SecurityException se) {
+            lawg.e("se: " + se);
+        }
+    }
+
+    @Override
+    public void onConnectionSuspended(int i) {
 
-        Bitmap icon = BitmapFactory.decodeResource(getResources(),
-                R.drawable.ic_darren);
+    }
 
-        Bitmap croppedIcon = getCroppedBitmap(icon);
+    @Override
+    public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
 
-        mCurrentMarker = mMap.addMarker(new MarkerOptions().position(latLng).title("Current Location").icon(BitmapDescriptorFactory.fromBitmap(croppedIcon)));
-      }
-    } catch (SecurityException se) {
-      lawg.e("se: " + se);
     }
-  }
 
-  @Override
-  public void onConnectionSuspended(int i) {
+    public Bitmap getCroppedBitmap(Bitmap bitmap) {
+        Bitmap output = Bitmap.createBitmap(bitmap.getWidth(),
+                bitmap.getHeight(), Bitmap.Config.ARGB_8888);
+        Canvas canvas = new Canvas(output);
+
+        final int color = 0xff424242; //Changing this to R.color.transparent removes icon
```

Comment:
Actually it can be any color, but when i used color.transparent it made the entire icon invisible